### DISCUSSION TITLE
sendParent fails if machine not spawned

### DISCUSSION THREAD

**Discussion by matthewpflueger** [statelyai/xstate#3191] [Q&A] [answered]
I have a simple machine that in my app is spawned but in my unit tests I only interpret the machine. Unfortunately, when I use sendParent in the simple machine it fails and my unit tests fail because, I'm guessing, my machine is only interpreted in the test (not spawned).

I thought about using actions.choose with a cond that checks if the machine has a parent before trying to use sendParent but for the life of me I can't figure out how to get at the parent inside the machine.

Any help or suggestions would be greatly appreciated!

Pseudo code of my problem:

```typescript
const myMachine = createMachine({
  initial: 'init',
  states: {
    init: {
     entry: sendParent('INITIALIZED') // This fails if myMachine is interpreted, okay if spawned
    }
  }
})

// In unit test I do:
interpret(myMachine)

// But in my app I do:
spawn(myMachine)
```

**Comment by davidkpiano** [comment, marked as answer]
What do you expect to happen if there is no parent? An error or just a warning and for the machine to silently continue?

One workaround you can try in your unit test is to provide a "fake" parent actor:

```js
interpret(myMachine, {
  parent: {
    send: () => void 0,
    subscribe: () => ({ unsubscribe: () => void 0 })
  }
});
```

**Reply by matthewpflueger** [reply]
Yeah, the fake parent is the better approach. 

What happens in a normal actor system if the reference is no longer valid? Would the message get dropped on the floor? Or a configure option to say we don't care if it gets delivered or not? My initial impression was that messages get dropped if the actor doesn't exist...

**Reply by davidkpiano** [reply]
Ideally it gets sent to a dead letter queue, but that is not implemented in XState yet (likely in v5).
